tests/timeseries/test_quantities.py

```diff
@@ -377,3 +377,23 @@ def test_abs() -> None:
     pct = Percentage.from_fraction(30)
     assert abs(pct) == Percentage.from_fraction(30)
     assert abs(-pct) == Percentage.from_fraction(30)
+
+
+def test_quantity_multiplied_with_precentage() -> None:
+    """Test the multiplication of all quantities with percentage."""
+    percentage = Percentage.from_percent(50)
+    power = Power.from_watts(1000.0)
+    voltage = Voltage.from_volts(230.0)
+    current = Current.from_amperes(2)
+    energy = Energy.from_kilowatt_hours(12)
+    percentage_ = Percentage.from_percent(50)
+
+    assert power * percentage == Power.from_watts(500.0)
+    assert voltage * percentage == Voltage.from_volts(115.0)
+    assert current * percentage == Current.from_amperes(1)
+    assert energy * percentage == Energy.from_kilowatt_hours(6)
+    assert percentage_ * percentage == Percentage.from_percent(25)
+
+    for quantity in [power, voltage, current, energy, percentage_]:
+        with pytest.raises(TypeError):
+            _ = quantity * 200.0
```

Comment:
I was thinking more like `Energy * Power` and `Energy * Energy`, etc.  Things that can have a `base_value` field.  I was thinking of this just because such a check was not there in the `__mul__` method (although you've now added it).